include/nuttx/mm/mm.h

```diff
@@ -88,9 +88,11 @@
 
 /* The kernel heap is never accessible from user code */
 
+#if defined(CONFIG_BUILD_KERNEL) || defined(CONFIG_BUILD_PROTECTED)
 #ifndef __KERNEL__
```

Comment:
```suggestion
#if !defined(__KERNEL__) && (defined(CONFIG_BUILD_FLAT)
```

Reply:
This suggestion is not equivalent. Is this what you intended to suggest?

```suggestion
#if !defined(__KERNEL__) && !defined(CONFIG_BUILD_FLAT)
```

Reply:
or merge into one line:
```
#if defined(__KERNEL__) && defined(CONFIG_BUILD_KERNEL) || defined(CONFIG_BUILD_PROTECTED)
```

Reply:
Sure, it's fine but we need indent `#ifndef __KERNEL` two spaces

Reply:
Gustavo, maybe it could be a good idea to put that comment:

CONFIG_MM_KERNEL_HEAP must be undefined only when building the Userspace image, which only exists under CONFIG_BUILD_KERNEL and CONFIG_BUILD_PROTECTED. So we need to ensure !defined(__KERNEL__)

 just before the #if testing to let people understand easily the intention

Reply:
Thanks @xiaoxiang781216 and @acassis for the feedback, I've implemented both suggestions. PTAL.